libbpfgo.go

```diff
@@ -339,7 +339,7 @@ func NewModuleFromFileArgs(args NewModuleArgs) (*Module, error) {
 	}
 
 	obj, errno := C.bpf_object__open_file(bpfFile, &opts)
-	if errno != nil {
+	if obj == nil && errno != nil {
```

Comment:
Why is checking errno not already enough to make sure there's an error? It seems that Go does clear errno before each C function call.

Reply:
Not related to the comment of @grantseltzer 
libbpf might return a non-null value in some cases.
For example: https://github.com/libbpf/libbpf/blob/master/src/libbpf.c#L7080.
In that case, we will mistakenly return that no error occured

Reply:
Even if Go does clear it before the call it's still possible that something within the call sets `errno` but we still don't actually have an error condition, the call that we originally made did succeed.

Without this change a few of the selftests error out. Also in this patch is a fix for map batch operations where `errno` could equal `ENOENT`. This happens when we've read an entire map, however we may still have gotten some data back so this patch fixes things to ensure whatever data we got is still returned, even if it's less than the batch amount.